src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +221,215 @@ pub fn print_lockfile_changes(
     }
 }
 
+pub fn upgrade_manifests(
+    ws: &mut Workspace<'_>,
+    opts: &UpdateOptions<'_>,
+) -> CargoResult<HashMap<String, Version>> {
+    let mut upgrades = HashMap::new();
+
+    if !opts.breaking {
+        return Ok(upgrades);
+    }
+
+    // Updates often require a lot of modifications to the registry, so ensure
+    // that we're synchronized against other Cargos.
+    let _lock = ws
+        .gctx()
+        .acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
+
+    let mut registry = PackageRegistry::new(opts.gctx)?;
+    registry.lock_patches();
+
+    for member in ws.members_mut() {
+        debug!("upgrading manifest for `{}`", member.name());
+
+        *member.manifest_mut().summary_mut() = member.manifest().summary().clone().try_map_dependencies(
+            |dependency| -> CargoResult<_> {
```

Comment:
Can we split this out?  For myself, I find overly long closures tend to make the code harder to read.